sktime/forecasting/arch.py

```diff
@@ -0,0 +1,319 @@
+        Name of the error distribution.  Currently supported options are:
+            * Normal: 'normal', 'gaussian' (default)
+            * Students's t: 't', 'studentst'
+            * Skewed Student's t: 'skewstudent', 'skewt'
+            * Generalized Error Distribution: 'ged', 'generalized error"
+    hold_back : int
+        Number of observations at the start of the sample to exclude when
+        estimating model parameters.  Used when comparing models with different
+        lag lengths to estimate on the common sample.
+    rescale : bool
+        Flag indicating whether to automatically rescale data if the scale
+        of the data is likely to produce convergence issues when estimating
+        model parameters. If False, the model is estimated on the data without
+        transformation.  If True, than y is rescaled and the new scale is
+        reported in the estimation results.
+
+    See Also
+    --------
+    Autoregressive Integrated Moving Average (ARIMA) models
+
+    References
+    ----------
+    .. [1] Jason Brownlee. How to Model Volatility with ARCH and GARCH for Time Series Forecasting.
+       https://machinelearningmastery.com/develop-arch-and-garch-models-for-time-series-forecasting-in-python/
+    .. [2] GitHub repository of arch package (soft dependency).
+       https://github.com/bashtage/arch
+    .. [3] Documentation of arch package (soft dependency). Forecasting Volatility with ARCH and it's variants.
+       https://arch.readthedocs.io/en/latest/univariate/introduction.html
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.arch import ARCH
+    >>> y = load_airline()
+    >>> forecaster = ARCH(
+    ...    mean='Constant',
+    ...    lags=0, vol='Garch',
+    ...    p=1,
+    ...    o=0,
+    ...    q=1,
+    ...    power=2.0,
+    ...    dist='Normal',
+    ...    hold_back=None,
+    ...    rescale=None)
+    >>> forecaster.fit(y)
+    ARCH(...)
+    >>> y_pred = forecaster.predict(fh=[1,2,3])
+    """
+
+    _tags = {
+        "scitype:y": "univariate",
+        "y_inner_mtype": ["np.ndarray", "pd.Series"],
+        "X_inner_mtype": ["np.ndarray", "pd.DataFrame"],
+        "requires-fh-in-fit": False,
+        "capability:pred_int": False,
+    }
+
+    def __init__(
+        self,
+        mean='Constant',
+        lags=0, vol='Garch',
+        p=1,
+        o=0,
+        q=1,
+        power=2.0,
+        dist='Normal',
+        hold_back=None,
+        rescale=None):
+
+        _check_soft_dependencies("arch", severity="error", object=self)
+
+        self.mean = mean
+        self.lags = lags
+        self.vol = vol
+        self.p = p
+        self.o = o
+        self.q = q
+        self.power = power
+        self.dist = dist
+        self.hold_back = hold_back
+        self.rescale = rescale
+        self._forecaster = None
+
+        super(ARCH, self).__init__()
+
+    def _fit(self, y, X=None, fh=None, **fit_params):
```

Comment:
the methods should not have parameters `**fit_params` etc, only the specified args

Reply:
Oh okay, but the `fit` method of the `arch` model takes some other parameters as well. What can I do about that?

Reply:
Thanks for letting me know about linting issues, may I know how I can handle multiple parameters of the `fit` method in the `arch` model. (currently **fit_params is used)